### PR TITLE
Fix `AmplitudeEstimation` is silently ignoring `is_good_state` in the estimation problem

### PR DESCRIPTION
**Pull request by Cryoris**
### Summary

As discussed in [Slack](https://qiskit.slack.com/archives/C7SS31917/p1680543628959579), the `AmplitudeEstimation` algorithm does not support `EstimationProblem.is_good_state` but just accept and silently ignores it, which leads to unexpected outcomes. This PR warns if the option is set and adds a note to the docs, that this option should be set differently.

### Details and comments

That `is_good_state` is not supported is a property of the algorithm itself, not laziness on the dev side 🙂 

I labelled this "bugfix" as I thought that silently ignoring arguments is a bug, but we could also label this as docs change.

### CODE REVIEW

**Review by ElePT — APPROVED** [review]
LGTM, but I would wait for Steve's approval too.

File: qiskit/algorithms/amplitude_estimators/ae.py
Comment (woodsp-ibm) [inline qiskit/algorithms/amplitude_estimators/ae.py:358]: As others could build other AE algorithms using the problem, rather than using a private var here to detect this, should we rather add some public method on the problem say. has_good_state to return if the user provided a good state callback or if is_good_state is just defaulting it? Maybe this here is an exception and using the private var with the same sub-package is ok.

Comment (Cryoris) [reply qiskit/algorithms/amplitude_estimators/ae.py:358]: Yeah it might be nice to allow querying that property. I'm not sure if it would be a bit overkill here, since we're not aware of any other use-cases so far, but we could still add it.

Comment (woodsp-ibm) [reply qiskit/algorithms/amplitude_estimators/ae.py:358]: It would avoid using a private member of the problem - I grant you some public method is unlikely to be used elsewhere, and the likelihood of the private implementation in the problem changing is minimal. Hence I concluded above with "Maybe this here is an exception and using the private var with the same sub-package is ok." I guess I'm still not so keen on it, but it works and if this is your preference...

Comment (Cryoris) [reply qiskit/algorithms/amplitude_estimators/ae.py]: Added in ee935b5 🙂

File: qiskit/algorithms/amplitude_estimators/ae.py
Comment (woodsp-ibm) [inline qiskit/algorithms/amplitude_estimators/ae.py:362]: >"The oracle has to be set using the grover_operator, otherwise "
>"a good state is identified by the objective qubits being in state 1.

I am trying to figure `otherwise` in that sentence. Should if I replace it by "where" ie

>"The oracle has to be set using the grover_operator, where "
>"a good state is identified by the objective qubits being in state 1.

Is this the same. Maybe the otherwise is trying to say it has some fixed good state determination that is other than (i.e. not) determined by the good_state function which is not supported and ignored. Maybe we should add "ignored" too.

So perhaps?...

The is_good_state function will be ignored. The AmplitudeEstimation class does 
not support an is_good_state function to identify good states. The oracle has to
be set using the grover_operator where a good state is identified by the objective
qubits being in state 1.

Comment (Cryoris) [reply qiskit/algorithms/amplitude_estimators/ae.py:362]: "Otherwise" is meant as in: If you want to set a custom oracle, you have to set it via the `grover_operator`. Otherwise (i.e. if no custom oracle is used) a the default oracle is used, which identifies good states as states, where all objective qubits are 1.

Comment (Cryoris) [inline qiskit/algorithms/amplitude_estimators/ae.py:362]:
```suggestion
                "The AmplitudeEstimation class does not support an is_good_state function to "
                "identify good states. For this algorithm, a custom oracle has to be encoded directly "
                "in the grover_operator. If no custom oracle is set, this algorithm identifies good "
                "states as those, where all objective qubits are in state 1."
```

Comment (Cryoris) [reply qiskit/algorithms/amplitude_estimators/ae.py]: I tried to make this clearer in 15b977a

**Review by woodsp-ibm — APPROVED** [review]

**Pull request merged**
